app/controllers/stripe-setup/bank-details/post.controller.js

```diff
@@ -18,24 +22,53 @@ module.exports = (req, res) => {
   const accountNumber = req.body[ACCOUNT_NUMBER_FIELD]
   const sortCode = req.body[SORT_CODE_FIELD]
 
-  const errors = validateBankDetails(accountNumber, sortCode)
+  let errors = validateBankDetails(accountNumber, sortCode)
   const pageData = {
     accountNumber,
     sortCode,
     errors
   }
 
   if (!lodash.isEmpty(errors)) {
-    return response.response(req, res, 'stripe-setup/bank-details/index', pageData)
+    return response(req, res, 'stripe-setup/bank-details/index', pageData)
   }
   if (req.body[ANSWERS_NEED_CHANGING_FIELD]) {
-    return response.response(req, res, 'stripe-setup/bank-details/index', pageData)
+    return response(req, res, 'stripe-setup/bank-details/index', pageData)
   }
   if (req.body[ANSWERS_CHECKED_FIELD]) {
-    // TODO: Stripe submission
-    return res.redirect(303, paths.dashboard.index)
+    return updateBankAccount(res.locals.stripeAccount.stripeAccountId, {
+      bank_account_sort_code: sortCode,
+      bank_account_number: accountNumber
+    })
+      .then(() => {
+        return connector.setStripeAccountSetupFlag(req.account.gateway_account_id, 'bank_account', req.correlationId)
+      })
+      .then(() => {
+        return res.redirect(303, paths.dashboard.index)
+      })
+      .catch(error => {
+        // check if it is Stripe related error
+        if (error.code) {
+          if (error.code === 'routing_number_invalid') {
+            // invalid sort code
+            errors = {}
+            errors[SORT_CODE_FIELD] = fieldValidationChecks.validationErrors.invalidSortCode
+            pageData.errors = errors
```

Comment:
i think this would be clearer if it were
```pageData.errors = { SORT_CODE_FIELD:  fieldValidationChecks.validationErrors.invalidSortCode }```
re-using the errors object makes it a bit hard to follow

Reply:
`SORT_CODE_FIELD` is constant which is evaluated as a property name, that is why we need `errors[SORT_CODE_FIELD]` which will be resolved to `errors { 'account-number': ... }`
I changed the controller similar way: `pageData.errors[SORT_CODE_FIELD] = ...` and made errors `const` again, thanks

Reply:
Ah yes you're right 👍